CMakeLists.txt

```diff
@@ -338,15 +338,20 @@ add_zeek_dynamic_plugin_build_interface_include_directories(
     ${PROJECT_SOURCE_DIR}/auxil/broker/libbroker
     ${PROJECT_SOURCE_DIR}/auxil/paraglob/include
     ${PROJECT_SOURCE_DIR}/auxil/rapidjson/include
+    ${PROJECT_SOURCE_DIR}/auxil/prometheus-cpp/core/include
     ${CMAKE_BINARY_DIR}/src
     ${CMAKE_BINARY_DIR}/src/include
     ${CMAKE_BINARY_DIR}/auxil/binpac/lib
-    ${CMAKE_BINARY_DIR}/auxil/broker/libbroker)
+    ${CMAKE_BINARY_DIR}/auxil/broker/libbroker
+    ${CMAKE_BINARY_DIR}/auxil/prometheus-cpp/core/include)
 
 # threading/formatters/JSON.h includes rapidjson headers and may be used
 # by external plugins, extend the include path.
 target_include_directories(zeek_dynamic_plugin_base SYSTEM
                            INTERFACE $<INSTALL_INTERFACE:include/zeek/3rdparty/rapidjson/include>)
+target_include_directories(
+    zeek_dynamic_plugin_base SYSTEM
+    INTERFACE $<INSTALL_INTERFACE:include/zeek/3rdparty/prometheus-cpp/include>)
```

Comment:
This might be opening a can of worms. We've previously attempted to remove rapidjson includes in public headers so that external plugins aren't exposed or aware of it. The rapidjson tweak above is meant to vanish with 7.1.

https://github.com/zeek/zeek/pull/3131

For OpenSSL, Dominik had done something similar
https://github.com/zeek/zeek/pull/3420

On first sight, this seems like a step towards exposing plugins to an external library again.

Also tangents the question if in the far future this dependency should from a distribution package ([fedora](https://packages.fedoraproject.org/pkgs/libprometheus-cpp/libprometheus-cpp-devel/fedora-40.html), [debian sid](https://packages.debian.org/sid/prometheus-cpp-dev)) rather than bundled.

Reply:
I realize hiding it likely results in a bunch of abstraction, so feel free to bounce.

Reply:
Shouldn't forward declarations in our headers suffice?

Reply:
This looks like a fix for https://github.com/zeek/zeek/issues/3760 btw.

Reply:
> Shouldn't forward declarations in our headers suffice?

The `Family` template type causes some issues. I was able to remove one set of forward declarations out of the Manager in 1a66c203005384975c1b38698cf929c746c3aca9, but the includes remain in the instrument types.

Reply:
Hmm, oh well - I'm going to close.

I did see that moving the `telemetry/*` includes from `src/OpaqueVal.h` to `Opaques.h` could be useful. It should fix existing `src/OpaqueVal.h` users.

Though it won't change the situation much if someone wants to use telemetry from a plugin.